src/luacheck/stages/check_table_fields.lua

```diff
@@ -437,7 +437,8 @@ local item_callbacks = {
    Cjump = ClosureState.handle_jump,
    Eval = ClosureState.handle_eval,
    Local = ClosureState.handle_local_or_set_item,
-   Set = ClosureState.handle_local_or_set_item
+   Set = ClosureState.handle_local_or_set_item,
+   OpSet = ClosureState.handle_local_or_set_item,
```

Comment:
@arichard4 I took a stab at both fixing the Lua error and integrating the test result changes from the last few releases into this branch. I wasn't 100% sure about this fix though, does it look sane to you?

Reply:
Fair enough. Feel free to reset to before my merge or rewrite/amend the merge to correct this or just add a new commit to adjust as appropriate.

Reply:
Pushed a revised version